File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        min_dis, min_idx = inf, -1
        for i, (px, py) in enumerate(points):
            if px == x or py == y:
                dis = abs(px - x) + abs(py - y)
                if dis < min_dis:
                    min_dis, min_idx = dis, i
        return min_idx

File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int smallest_distance = Integer.MAX_VALUE, smallest_idx = -1;
        for(int i = 0; i < points.length; i++){
            int[] point = points[i];
            if(point[0] != x && point[1] != y)
                continue;
            
            int distance = Math.abs(point[0] - x) + Math.abs(point[1] - y);
            if(distance < smallest_distance){
                smallest_distance = distance;
                smallest_idx = i;
            }
        }
        return smallest_idx;
    }
}